### PR TITLE
OSOE-796: Fix setup-sql-server GitHub Windows Larger Runner incompatibility in Lombiq.GitHub.Actions

### PR DESCRIPTION
**Pull request by Psichorex**
[OSOE-796](https://lombiq.atlassian.net/browse/OSOE-796)

[OSOE-796]: https://lombiq.atlassian.net/browse/OSOE-796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/build-and-test-windows.yml [hunk 1]
```diff
@@ -33,13 +33,13 @@ jobs:
     if: github.ref_name != github.event.repository.default_branch &&
         (github.event_name == 'workflow_dispatch' || github.event.label.name == 'run-windows-build')
     name: Build and Test Windows - root solution (larger runners)
-    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@dev
+    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@issue/OSOE-796
     with:
       parent-job-name: "root-solution-larger-runners"
       # Should be windows-2022-8core but temporarily isn't due to this bug: https://github.com/Lombiq/GitHub-Actions/issues/320.
```
Comment (Piedone) [inline .github/workflows/build-and-test-windows.yml]: The comment is outdated and `dotnet-test-process-timeout` is not reverted.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-test-windows.yml (hunk 1 above)
Comment (Psichorex) [reply .github/workflows/build-and-test-windows.yml]: It failed me once saying it exceeded 360000 so thats why I left that

Comment (Piedone) [reply .github/workflows/build-and-test-windows.yml]: The point of using a larger runner is not to have that. It should be faster. Perhaps not that, but then you need to tune the number.

**Comment by Psichorex** [comment]
UI Tests are flaky here so I am investigating what could be causing this. 
https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/7820218881?pr=694

**Comment by Piedone** [comment]
That's more of this issue, unfortunately: https://github.com/Lombiq/GitHub-Actions/blob/8a0d9bf1a0025857fb42f5903f516db8401e4166/.github/actions/test-dotnet/action.yml#L121 See https://github.com/actions/upload-artifact/issues/240. Not because of your changes.

File: .github/workflows/build-and-test-windows.yml (hunk 1 above)
Comment (Psichorex) [reply .github/workflows/build-and-test-windows.yml]: I tried up to `640000` and it still failed. It probably should be way faster if tuning is needed so I am guessing the tuning shouldn't be from `720000` -> `700000`. Other than this I addressed everything.

Comment (Piedone) [reply .github/workflows/build-and-test-windows.yml]: It works though: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/7907374782

**Pull request merged**